### ISSUE TITLE
Consider adding dependency for `thiserror`

### ISSUE THREAD

**Issue by jhdcs** [ros2-rust/ament_cargo_plugin#3]
The `thiserror` crate (https://github.com/dtolnay/thiserror) provides some nice features for quickly generating descriptive errors. It's meant to be used in library crates, which I think that this could be safely counted as.

I know we haven't written any errors yet, but I wanted to bring this up _before_ we do, since it's much easier to implement something from the very beginning, rather than re-tooling a codebase later.

What are your thoughts on including this crate?

**Comment by esteve** [comment]
Frankly, just use whatever crates are available and make your life easier, we'll get to discuss and review the code before it's merged.

**Comment by bergercookie** [comment]
Yes, agree with using `thiserror`. Basically the convention I'm following is using `thiserror` for library crates and [anyhow](https://github.com/dtolnay/anyhow)  for application code.
